framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -637,6 +638,14 @@ private void checkContractsAtMethodDeclaration(
                         FlowExpressionContext.buildContextForMethodDeclaration(
                                 node, getCurrentPath(), checker.getContext());
             }
+
+            DependentTypesHelper dependentTypesHelper = atypeFactory.getDependentTypesHelper();
```

Comment:
This block of code (the next 6 lines) happens quite a few times in this and other files.  Consider abstracting it out, so that it's a single method call in all the places it occurs.